lib/swaps/Swaps.ts

```diff
@@ -571,13 +560,15 @@ class Swaps extends EventEmitter {
 
       const makerClientCltvDelta = this.swapClientManager.get(makerCurrency)!.cltvDelta;
       this.logger.debug(`maker client CLTV delta: ${makerClientCltvDelta}`);
-      const takerClientCltvDelta = this.swapClientManager.get(takerCurrency)!.cltvDelta;
-      this.logger.debug(`taker client CLTV delta: ${takerClientCltvDelta}`);
 
-      // cltvDelta can't be zero for swap clients (checked in constructor)
-      const cltvDeltaFactor = makerClientCltvDelta / takerClientCltvDelta;
+      /** The ratio of the average time for blocks on the taker (2nd leg) currency per blocks on the maker (1st leg) currency. */
+      const cltvDeltaFactor = takerSwapClient.minutesPerBlock / makerSwapClient.minutesPerBlock;
       this.logger.debug(`CLTV delta factor: ${cltvDeltaFactor}`);
 
+      // Here we calculate the minimum CLTV delay the maker will expect on the final hop to him on
+      // the first leg of the swap. This is equal to the maker's configurable cltvDelta plus the
+      // total delay of the taker route times a factor to convert taker blocks to maker blocks.
+      // This is to ensure that the 1st leg payment HTLC doesn't expire before the 2nd leg.
       deal.makerCltvDelta = makerClientCltvDelta + Math.ceil(routeCltvDelta * cltvDeltaFactor);
```

Comment:
The math here should be : 100+ 144*40

Reply:
The first number is the maker's configured cltvDelta value (the config option we're considering renaming something like `lockDuration`), then it's the total delay of the second leg (which is always 100 for raiden, variable for lnd) * the block time factor which in the case of raiden to bitcoin is 40. So I think this is right pending some clarification on that first value.